src/shared/test/download-file-test.js

```diff
@@ -18,26 +18,46 @@ describe('download-json-file', () => {
         removeChild: sinon.stub(),
       },
     };
-  });
 
-  it('generates export file with provided annotations', () => {
-    const filename = 'my-file.json';
-    const data = { foo: ['bar', 'baz'] };
-
-    const fileContent = downloadJSONFile(data, filename, fakeDocument);
+    sinon.spy(window, 'Blob');
+  });
 
-    assert.equal(fileContent, JSON.stringify(data, null, 2));
+  afterEach(() => {
+    window.Blob.restore();
+  });
 
+  function assertDownloadHappened(filename, fileContent, type) {
     assert.calledWith(fakeDocument.createElement, 'a');
     assert.calledWith(fakeDocument.body.appendChild, fakeLink);
     assert.calledWith(fakeDocument.body.removeChild, fakeLink);
 
+    assert.calledWith(window.Blob, [fileContent], { type });
+
     assert.calledWith(
       fakeLink.setAttribute.firstCall,
       'href',
       sinon.match.string,
     );
     assert.calledWith(fakeLink.setAttribute.secondCall, 'download', filename);
     assert.equal('hidden', fakeLink.style.visibility);
+  }
```

Comment:
Another way this could have worked is to stub `HTMLElement.prototype.click` during the test and then verify that it is called on an element which a) is connected and b) has the `download` attribute set and c) has the expected content in the URL.